extensions/wikia/TemplateClassification/maintenance/classifyNonArticleTemplates.class.php

```diff
@@ -0,0 +1,127 @@
+			return false;
+		}
+
+		$this->setDB( wfGetDB( DB_SLAVE ) );
+		$tcs = new TemplateClassificationService();
+
+		$namespacesTemplates = ( new \WikiaSQL() )
+			->SELECT( 'pt.page_title as template_title, pt.page_id as template_id, p.page_id as page_id' )
+			->FROM( 'templatelinks' )->AS_( 'tl' )
+			->INNER_JOIN( 'page' )->AS_( 'pt' )
+			->ON( 'tl.tl_title', 'pt.page_title' )
+			->INNER_JOIN( 'page' )->AS_( 'p' )
+			->ON( 'tl.tl_from', 'p.page_id' )
+			->WHERE( 'tl.tl_namespace' )->EQUAL_TO( NS_TEMPLATE )
+			->AND_( 'pt.page_namespace' )->EQUAL_TO( NS_TEMPLATE )
+			->AND_( 'p.page_namespace' )->IN( $wgContentNamespaces )
+			->runLoop( $this->getDB(), function ( &$pages, $row ) {
+				if ( !isset( $pages[$row->template_id]['title'] ) ) {
+					// First run for this template ID
+					$pages[$row->template_id]['title'] = $row->template_title;
+					$pages[$row->template_id]['linkingPages'] = [];
+				}
+
+				$pages[$row->template_id]['linkingPages'][] = (int)$row->page_id;
+			} );
+
+		$countFirstLevelTemplates = 0;
+		foreach ( $namespacesTemplates as $templateId => $data ) {
+			$templateTitle = $data['title'];
+			$linkingPages = $data['linkingPages'];
+			$count = count( $linkingPages );
+
+			$this->out( "\nProcessing {$count} inclusions of template {$templateTitle} ({$templateId})" );
+			$isFirstLevel = false;
+
+			foreach ( $linkingPages as $pageId ) {
+				$pageRawTextLc = strtolower( WikiPage::newFromID( $pageId )->getRawText() );
+				$templateTitleLc = strtolower( $templateTitle );
+				if ( strpos( $pageRawTextLc, "{{{$templateTitleLc}" ) > -1 ) {
+					$countFirstLevelTemplates++;
+					$isFirstLevel = true;
+					break;
+				}
+			}
+
+			if ( $isFirstLevel ) {
+				$this->out( "{$templateTitle} - First level inclusion found in {$pageId}!" );
+			} else {
+				$this->out( "{$templateTitle} is just a nested template! Classify it!" );
+				if ( !$this->dryRun ) {
+					try {
+						$tcs->classifyTemplate(
+							$wgCityId,
+							$templateId,
+							TemplateClassificationService::TEMPLATE_NOT_ART,
+							self::NONARTICLE_MAINTENANCE_PROVIDER,
+							self::NONARTICLE_MAINTENANCE_ORIGIN
+						);
+					} catch ( \Swagger\Client\ApiException $e ) {
+						$this->out( 'Classification failed!' );
+					}
+				}
+			}
+		}
+
+		$countContentNsTemplates = count( $namespacesTemplates );
+		$results = [
+			'wiki_id' => $wgCityId,
+			'dbname' => $wgDBname,
+			'Templates in content NS' => $countContentNsTemplates,
+			'First-level templates' => $countFirstLevelTemplates,
+			'Nested templates' => $countContentNsTemplates - $countFirstLevelTemplates,
+		];
+
+		$this->out( "\nResults: " . json_encode( $results ) );
+		$this->logResults( $results );
+	}
+
+	private function out( $s ) {
+		if ( !$this->quiet ) {
+			$this->output( "$s\n" );
+		}
+	}
+
+	private function logResults( array $results ) {
+		if ( isset( $this->logFile ) ) {
+			$data = implode( ',', array_values( $results ) );
```

Comment:
`array_values` is unneeded here. `implode( ',', array_values( $results ) ) === implode( ',', $results )`